ppl/archive/index/dirwriter.go

```diff
@@ -0,0 +1,80 @@
+package index
+
+import (
+	"context"
+	"path"
+
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zqe"
+	"go.uber.org/multierr"
+)
+
+type DirWriter []*Writer
+
+func NewDirWriter(ctx context.Context, d Dir, defs []*Def) (DirWriter, error) {
+	writers := make(DirWriter, 0, len(defs))
+	for _, def := range defs {
+		w, err := d.newIndexWriter(ctx, def)
+		if err != nil {
+			if zqe.IsExists(err) {
+				// skip indices that already exist
+				continue
+			}
+
+			writers.Abort()
+			return nil, err
+		}
+
+		writers = append(writers, w)
+	}
+	return writers, nil
+}
+
+func (ws DirWriter) WriteBatch(batch zbuf.Batch) error {
+	for _, w := range ws {
+		batch.Ref()
+		if err := w.WriteBatch(batch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ws DirWriter) Write(rec *zng.Record) error {
+	for _, w := range ws {
+		if err := w.Write(rec); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ws DirWriter) Close() (merr error) {
+	for _, w := range ws {
+		if err := w.Close(); err != nil {
+			merr = multierr.Append(merr, err)
+		}
+	}
+	// Drop all indices if there was an error closing.
+	if merr != nil {
+		ws.Abort()
+	}
```

Comment:
This is probably not what we want, since this will end up calling Abort on already closed indices. I don't think we should do anything extra if there's a close error on the index writer.

```suggestion
```